### PR TITLE
properly deal with scrolling for RenderDraw in xyz, video

### PR DESCRIPTION
**Pull request by rcoreilly**
xyz was not scrolling properly due to logic issue in direct render draw, now fixed.

### CODE REVIEW

File: video/video.go [hunk 1]
```diff
@@ -84,8 +84,21 @@ func (v *Video) RenderDraw(drw system.Drawer, op draw.Op) {
...
+		ibb.Min.X = tbb.Min.X - pos.X
+	}
+	if pos.Y < tbb.Min.Y {
+		ibb.Min.X = tbb.Min.Y - pos.X
+	}
+	sz := ibb.Size()
+	if sz.X <= 0 || sz.Y <= 0 {
+		return
+	}
+	drw.Scale(bb, frame, ibb, v.Rotation, draw.Src, unchanged)
```
Comment (kkoreilly) [inline video/video.go]: Is there a way to avoid forcing each widget to do all of this work manually?

Comment (rcoreilly) [reply video/video.go]: making a helper function..  could do that.

File: core/layout.go [hunk 2]
```diff
@@ -1847,3 +1849,25 @@ func (fr *Frame) scrollResetIfNone() {
 		}
 	}
 }
+
+// DirectRenderDrawBBoxes returns the destination and source bounding boxes
+// for RenderDraw call for widgets that do direct rendering.
+// The destBBox.Min point can be passed as the dp destination point for Draw
+// function, and srcBBox is
+func (wb *WidgetBase) DirectRenderDrawBBoxes(srcFullBBox image.Rectangle) (destBBox, srcBBox image.Rectangle, empty bool) {
```
Comment (kkoreilly) [inline core/layout.go:1858]: Your documentation trails off here. Also I thought we weren't using `DirectRender` terminology anymore?

**Review by kkoreilly — CHANGES_REQUESTED** [review]

File: core/layout.go (hunk 2 above)
Comment (rcoreilly) [reply core/layout.go:1858]: oops. DirectRender is still used for direct renderers -- but everyone just has a RenderDraw method.

**Review by kkoreilly — APPROVED** [review]

**Pull request merged**